weedcoco/repo/deposit.py

```diff
@@ -223,13 +228,44 @@ def get_all_existing_hash(self, repository):
             if path.split("/")[0] == "images"
         }
 
-    def create_image_hash(self):
+    def create_image_hash_and_mapping(self):
+        annotations_hash = self.create_annotation_hash()
+        self.create_image_hash(annotations_hash)
+
+    def create_annotation_hash(self):
+        annotations_hash = defaultdict(str)
+        with open(self.weedcoco_path) as f:
+            weedcoco = json.load(f)
+        image_id_mapping = {
+            image["id"]: image["file_name"].split("/")[-1]
+            for image in weedcoco["images"]
+        }
+        image_annotation_mapping = {
+            image_id_mapping[annotation["image_id"]]: {
+                key: value
+                for key, value in annotation.items()
+                if key in COCO_ANNOTATION_FIELD
+            }
+            for annotation in weedcoco["annotations"]
+        }
+        for image_name, image_annotation in image_annotation_mapping.items():
+            annotations_hash[image_name] = hashlib.md5(
+                json.dumps(image_annotation, sort_keys=True).encode("utf-8")
+            ).hexdigest()
+        return annotations_hash
+
+    def create_image_hash(self, annotations_hash):
```

Comment:
I like that we're keeping the image hash and annotation hash separate - it means we don't lose the ability to check for duplicate images with different annotations